src/sst/core/eli/elibase.h

```diff
@@ -124,28 +122,27 @@ combineEliInfo(std::vector<T>& base, const std::vector<T>& add)
 
 struct LibraryLoader
 {
-    virtual void load() = 0;
-    virtual ~LibraryLoader() {}
+    virtual void load()      = 0;
+    virtual ~LibraryLoader() = default;
 };
 
 class LoadedLibraries
 {
 public:
-    using InfoMap    = std::map<std::string, std::list<LibraryLoader*>>;
+    using InfoMap    = std::map<std::string, std::deque<std::shared_ptr<LibraryLoader>>>;
```

Comment:
`std::deque` is faster than `std::list` and `std::vector` for adding an unknown number of items repeatedly to the end of a list which grows. `std::list` requires more dynamic memory allocations and is cache-unfriendly, and `std::vector` requires moving the entire vector's data to a new location if it has to increase its capacity. `std::deque` is also randomly accessible like `std::vector`, but it's not contiguous. I would look at all places `std::vector` and `std::list` are used and see if `std::deque` might be better.